### PR TITLE
Getting minimum utxo

### PR DESCRIPTION
**Pull request by paweljakubas**
# Issue Number

#1893 

# Overview

- [x] I have extended `ProtocolParameters` and `ApiNetworkParameters`
- [x] I have updated swagger
- [x] I have used minimumUtxoValue in feeOpts in determination of dustThreshold
- [x] I have updated testing
- [x] I have handled migration

# Comments

### CODE REVIEW

**Comment by paweljakubas** [comment]
bors try

File: lib/core/src/Cardano/Wallet.hs [hunk 1]
```diff
@@ -1283,13 +1288,14 @@ selectCoinsForMigrationFromUTxO
...
+    -> W.Coin
     -> WalletId
        -- ^ The source wallet ID.
     -> ExceptT ErrSelectForMigration IO [CoinSelection]
-selectCoinsForMigrationFromUTxO ctx utxo txp wid = do
+selectCoinsForMigrationFromUTxO ctx utxo txp minUtxo wid = do
     let feePolicy@(LinearFee (Quantity a) _ _) = txp ^. #getFeePolicy
-    let feeOptions = (feeOpts tl Nothing feePolicy)
-            { dustThreshold = Coin $ ceiling a }
+    let minUtxo' = max (Coin $ ceiling a) minUtxo
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet.hs:1297]: :+1:

File: lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs [hunk 2]
```diff
@@ -152,6 +152,7 @@ ProtocolParameters
     protocolParametersTxMaxSize             Word16          sql=tx_max_size
     protocolParametersDecentralizationLevel Percentage      sql=decentralization_level
     protocolParametersDesiredNumberOfPools  Word16          sql=desired_pool_number
+    protocolParametersMinimumUtxoValue      W.Coin          sql=minimum_utxo_value
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs:155]: We're going to need a migration for this one with a default value.

**Review by KtorZ — COMMENTED** [review]
Looks good as a first step, I wonder how well it's going to work with the coin selection.

Beware that we need a migration for the added column!

File: lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs (hunk 2 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs:155]: cherry-picked one commit and also handled the migration in https://github.com/input-output-hk/cardano-wallet/pull/1894/commits/487bb4f92fe88eff3f38cd02107337e9735ac8c3

**Comment by paweljakubas** [comment]
bors try

**Review by KtorZ — APPROVED** [review]

**Pull request merged**